packages/core/src/utils.ts

```diff
@@ -446,6 +446,43 @@ function deepEqualsObject(
   return true;
 }
 
+/**
+ * Checks if object2 includes all fields and values of object1.
+ * It doesn't matter if object2 has extra fields.
+ * @param object1 The first object.
+ * @param object2 The second object.
+ * @returns True if object2 includes all fields and values of object1.
+ */
+export function deepIncludes(object1: unknown, object2: unknown): boolean {
+  if (isEmpty(object1)) {
+    return true;
+  }
+  if (isEmpty(object2)) {
+    return false;
+  }
+  if (Array.isArray(object1) && Array.isArray(object2)) {
+    return deepIncludesArray(object1, object2);
+  }
+  if (Array.isArray(object1) || Array.isArray(object2)) {
+    return false;
+  }
+  if (isObject(object1) && isObject(object2)) {
+    return deepIncludesObject(object1, object2);
+  }
+  if (isObject(object1) || isObject(object2)) {
+    return false;
+  }
+  return object1 === object2;
+}
+
+function deepIncludesArray(array1: any[], array2: any[]): boolean {
+  return array1.every((value, index) => deepIncludes(value, array2[index]));
```

Comment:
This is going to require that the array elements being in the same order, which [the FHIR spec does not require](https://hl7.org/fhir/elementdefinition-examples.html#pattern-array).  Instead, this should just verify that every value from `array1` is in `array2`, e.g. `array1.every((value) => array2.includes(value))`

Reply:
The .includes will return false even if the objects are the same. I ended up calling deepEquals for this as a check